shared/js/background/amp-protection.es6.js

```diff
@@ -0,0 +1,183 @@
+        return false
+    }
+
+    ampSettings = tdsStorage.config.features.ampLinks.settings
+
+    return true
+}
+
+/**
+ * This method checks if the given url is a Google hosted AMP url and resturns the canonical url if it is.
+ *
+ * @param {Site} site - the initiating site
+ * @param {string} url - the url being loaded
+ * @returns canonical url if found, null otherwise
+ */
+function extractAMPURL (site, url) {
+    if (!ensureConfig()) {
+        return null
+    }
+
+    if (site.specialDomainName || !site.isFeatureEnabled(featureName)) {
+        return null
+    }
+
+    for (const regexPattern of ampSettings.linkFormats) {
+        const match = url.match(regexPattern)
+        if (match && match.length > 1) {
+            const newSite = new Site(match[1].startsWith('http') ? match[1] : `https://${match[1]}`)
+
+            if (newSite.specialDomainName || !newSite.isFeatureEnabled(featureName)) {
+                return null
+            }
+
+            return newSite.url
+        }
+    }
+
+    return null
+}
+
+/**
+ * Check if the given request is a suspected 1st party AMP URL. If it is, then we will attempt to fetch the canonical URL
+ *
+ * @param {object} requestData - request data
+ * @param {object} thisTab - current tab
+ * @param {object} mainFrameRequestURL - main frame request url
+ * @returns true if the request is a suspected 1st party AMP url
+ */
+function tabNeedsDeepExtraction (requestData, thisTab, mainFrameRequestURL) {
+    if (utils.getBrowserName() !== 'moz') {
+        // deep extraction is only supported on Firefox
+        return false
+    }
+
+    if (requestIsExtension(requestData)) {
+        return false
+    }
+
+    if (thisTab.ampUrl && thisTab.ampUrl === mainFrameRequestURL.href) {
+        return false
+    }
+
+    return isAMPURL(mainFrameRequestURL.href)
+}
+
+/**
+ * This method checks if the given url is a 1st party AMP url
+ *
+ * @param {string} url - the url being loaded
+ * @returns true is the url is suspected to be a 1st party AMP url
+ */
+function isAMPURL (url) {
+    if (!ensureConfig()) {
+        return false
+    }
+
+    const site = new Site(url)
+    if (site.specialDomainName || !site.isFeatureEnabled(featureName)) {
+        return false
+    }
+
+    return ampSettings.keywords.some(keyword => url.includes(keyword))
+}
+
+/**
+ * This async method will fetch the DOM content of a suspected 1st party AMP URL and will
+ * return the canonical URL if one is found.
+ *
+ * @param {Site} site - the initiating site
+ * @param {string} url - the url being loaded
+ * @returns cancalon url if found, null otherwise
+ */
+async function fetchAMPURL (site, url) {
+    if (!ensureConfig()) {
+        return null
+    }
+
+    if (site.specialDomainName || !site.isFeatureEnabled(featureName) || site.url === 'about:blank') {
+        return null
+    }
+
+    let data = null
+    const timeoutController = new AbortController()
+    setTimeout(() => timeoutController.abort(), 1500)
```

Comment:
Can we make timeout constant this a config setting as it could be a source of breakage / need tweaking?